### ISSUE TITLE
First Image size is smaller then other Images

### ISSUE THREAD

**Issue by nitinjiniguru** [spatie/pdf-to-image#67]
I have used this extension for converting the pdf to image

it converts perfectly every image but only issue is that first image is of the half resolution of the others

what should be the problem??

each and every variables are declared proper and also receives the values as well

**Comment by freekmurze** [comment]
Do you have some example code for me to look at?

**Comment by nitinjiniguru** [comment]
`if(is_file($pathToPdf)){
			$pdf = new Pdf($pathToPdf); 
			$pageCount = $pdf->getNumberOfPages(); //returns an int  
			for ($i = 1; $i <= $pageCount; $i++) { 
				$pathToWhereImageShouldBeStored = $path."/page_".$i.".png";
				array_push($allImages,$relativePath."/page_".$i.".png"); 
				$pdf->setPage($i)->setOutputFormat('png')->saveImage($pathToWhereImageShouldBeStored);
			}
			unlink($pathToPdf); // remove file from server
			return json_encode($allImages);
		}else{
			return json_encode($allImages);
		}`

**Comment by nitinjiniguru** [comment]
![codeofmyapp](https://user-images.githubusercontent.com/29449700/29814082-be27f33c-8cc9-11e7-89c4-ab8ec244a227.jpg)
this my code and i got solution from [https://github.com/spatie/pdf-to-image/issues/37#issuecomment-325567722](url) 
i have commented in vendor/spatie/pdf-to-image/src/pdf.php 

and it works for me.

so as vendor folder we do not commit generally so i change it and kept on my app and now it works

but if you have any solution from implementation than thats good

**Comment by fosron** [comment]
+1 

Page 1 - 595 × 842 pixels
Page 2  -2479 × 3508 pixels

**Comment by freekmurze** [comment]
I'll try to look at this soon. I'd appreciate a fully tested PR that fixes this.

**Comment by joshuadegier** [comment]
I've tested a few PDFs locally using this code:

```php
<?php

require 'vendor/autoload.php';

$files = [
    'example.pdf',
    'example2.pdf',
    'example3.pdf',
];

foreach($files as $file) {

    echo "{$file}<br />";

    $pdf = new Spatie\PdfToImage\Pdf($file);

    for($i = 1; $i<=$pdf->getNumberOfPages(); $i++)
    {
        $pdf->setPage($i)
            ->setOutputFormat('png')
            ->saveImage('page-' . $i . '.png');

        list($width, $height, $type, $attr) = getimagesize('page-' . $i . '.png');
        echo "Image {$i}: w:{$width} h:{$height}<br>";
    }

    echo "<hr />";
}
```

The output is as expected:

```
example.pdf
Image 1: w:1191 h:1684
Image 2: w:1191 h:1684
example2.pdf
Image 1: w:1190 h:1684
Image 2: w:1190 h:1684
Image 3: w:1190 h:1684
Image 4: w:1190 h:1684
example3.pdf
Image 1: w:1190 h:1684
Image 2: w:1190 h:1684
```

I think the problem lies in the PDFs that are being converted. There might be issues with different DPI and/or resolutions of the frontpage of these files. Any possibility you share a file that has issues?

**Comment by BrentRobert** [comment]
Could not reproduce either, resolution stays the same for all tested pdf's.

**Comment by morloderex** [comment]
Or could it be an issue with different versions of imagemagick?

**Comment by blitsoftwares** [comment]
+1

I'm creating a pdf image with 3 pages. The 3 generated images are the last page of the pdf.

`$pdf = new \Spatie\PdfToImage\Pdf($this->documentFile->getUrl());`
`$pdf->setPage($this->page)->setOutputFormat('jpg')->saveImage(public_path('crns_ocr/'.$path) ."/". $this->documentFile->hash_file. "_pg" . $this->page . '.jpg');`

It ignores the page number in setPage(integer).
Regardless of the pdf and the number of pages, always generate the last page. If the pdf has 10 pages, the 10 images will be the last page of the pdf.

Help!

**Comment by freekmurze** [comment]
Closing due to inactivity. Cannot reproduce this issue.

**Comment by vtsoft** [comment]
the problem occurs to those who uses the spatie/pdf-to-image version 1.4. It seems that in pdf.php class, it's lacking the following functions below:

setImageResolution and 
resampleImage.

we cannot upgrade to a version higher than 1.4 since our application is using php version 5.5 and versions  above 1.4 is only available for php versions 7.2 and 7.3+. And I think it's where you apply the fix.

try setting the environment to php 5.5 and spatie/pdf-to-image 1.4 and you can reproduce the issue.

I hope that your latest version of spatie/pdf-to-image will become backward compatible with the old php versions. maybe just start with php version 5.
